hnn_core/tests/test_network.py

```diff
@@ -112,6 +112,25 @@ def test_spikes(tmpdir):
 
     spikes = Spikes(times=spiketimes, gids=spikegids, types=spiketypes)
 
+    with pytest.raises(TypeError, match="indices must be int, slice, or "
+                       "array-like, not str"):
+        spikes['1']
+
+    with pytest.raises(TypeError, match="indices must be int, slice, or "
+                       "array-like, not float"):
+        spikes[1.0]
+
+    with pytest.raises(ValueError, match="ndarray cannot exceed 1 dimension"):
+        spikes[np.array([[1, 2], [3, 4]])]
+
+    with pytest.raises(TypeError, match="gids must be of dtype int, "
+                       "not float64"):
+        spikes[np.array([1, 2, 3.0])]
+
+    with pytest.raises(TypeError, match="gids must be of dtype int, "
+                       "not float64"):
+        spikes[[0, 1, 2, 2.0]]
```

Comment:
maybe also some smoke test that checks that `vsoma` > -65 or whatever makes sense?

Reply:
Really good idea! Most comprehensive test would be to check that voltages above a certain threshold align with spike times. Perhaps that can be added to a future PR, with just the baseline check for now.

Important to note that vsoma is only initialized after a simulation run. This could be added to test_dipole() at the risk of poor file naming. I'd prefer not to run an extra sim in this file as it would slow down the test.

Reply:
An option would be to run a reduced (25 ms) simulation here. However, in the interest of comparing all outputs for similarity to HNN (@rythope), `test_compare_across_backends` would be a good place to check the vsoma time-series, similar to dipole.
https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/tests/test_parallel_backends.py#L62

Reply:
@ntolley I think you should use `pytest` fixtures for avoiding multiple simulations. And @blakecaldwell 's idea is good too. Let me know if you have trouble figuring it out. I can point you to some example code.

Reply:
Also okay to add the second test in another PR if you prefer. We need to keep productive contributors motivated !